crates/swift-bridge-ir/src/codegen/codegen_tests/opaque_rust_type_codegen_tests.rs

```diff
@@ -527,3 +527,82 @@ void* __swift_bridge__$Foo$new(void);
         .test();
     }
 }
+
+/// Verify that we can use a Swift class with a throwing init.
+mod extern_rust_class_with_throwing_init {
+    use super::*;
+
+    fn bridge_module_tokens() -> TokenStream {
+        quote! {
+            mod foo {
+                enum SomeErrEnum {
+                    Variant1,
+                    Variant2,
+                }
+                extern "Rust" {
+                    type Foo;
+
+                    #[swift_bridge(init)]
+                    fn new() -> Result<Foo, SomeErrEnum>;
+                }
+            }
+        }
+    }
+
+    fn expected_rust_tokens() -> ExpectedRustTokens {
+        ExpectedRustTokens::Contains(quote! {
+            # [export_name = "__swift_bridge__$Foo$new"]
+            pub extern "C" fn __swift_bridge__Foo_new() -> ResultFooAndSomeErrEnum{
+                match super :: Foo :: new() {
+                    Ok(ok) => ResultFooAndSomeErrEnum::Ok(Box::into_raw(Box::new({
+                        let val: super::Foo = ok;
+                        val
+                    })) as *mut super::Foo),
+                    Err(err) => ResultFooAndSomeErrEnum::Err(err.into_ffi_repr()),
+                }
+            }
+        })
+    }
+
+    const EXPECTED_SWIFT: ExpectedSwiftCode = ExpectedSwiftCode::ContainsAfterTrim(
+        r#"
+public class Foo: FooRefMut {
+    var isOwned: Bool = true
+
+    public override init(ptr: UnsafeMutableRawPointer) {
+        super.init(ptr: ptr)
+    }
+
+    deinit {
+        if isOwned {
+            __swift_bridge__$Foo$_free(ptr)
+        }
+    }
+}
+extension Foo {
+    public convenience init() throws {
+        let val = __swift_bridge__$Foo$new(); switch val.tag { case __swift_bridge__$ResultFooAndSomeErrEnum$ResultOk: self.init(ptr: val.payload.ok) case __swift_bridge__$ResultFooAndSomeErrEnum$ResultErr: throw val.payload.err.intoSwiftRepr() default: fatalError() }
```

Comment:
Let's use an `if` statement instead of a `switch` so that we don't need this `default: fatalError()`

The `default: fatalError` makes it seem like this can fail, even though it can't.

Reply:
Done. However, I don't address the others yet. 

For example:
https://github.com/chinedufn/swift-bridge/blob/495611ba39bc6b08838408751b92f6329b98f321/crates/swift-bridge-ir/src/bridged_type/bridgeable_result.rs#L247

I'll address this in the next PR, or we can open it as good first issue.